app/apps/route.js

```diff
@@ -3,6 +3,11 @@ import Ember from 'ember';
 export default Ember.Route.extend({
   model: function(){
     var stack = this.modelFor('stack');
-    return stack.get('apps');
+    var apps = stack.get('apps');
+    if (apps.get('isFulfilled')) {
```

Comment:
This gets the state of the app's `ArrayController` promise.  `isFulfilled` being true indicates the promise is resolved and apps already live on the client, thus the need to `reload()` them to get latest.   Any other states of the promise would indicate the request is still in flight, so it can be returned assuming it's the latest data.  (I had to look this up to figure it out, so correct me if I am wrong @mixonic)